models.py

```diff
@@ -26,86 +28,32 @@ class Status(IntEnum):
     FINISHED = 2
 
 
-class MaintenanceWindow:
+class MaintenanceWindow(BaseModel):
     """Class to store a maintenance window."""
 
-    def __init__(self, start, end, controller, **kwargs):
-        """Create an instance of MaintenanceWindow.
-
-        Args:
-            start(datetime): when the maintenance will begin
-            end(datetime): when the maintenance will finish
-            items: list of items that will be maintained;
-                each item can be either a switch, a link or a client interface
-        """
-        # pylint: disable=invalid-name
-        self.controller = controller
-        items = kwargs.get('items')
-        if items is None:
-            items = []
-        mw_id = kwargs.get('mw_id')
-        self.id = mw_id if mw_id else uuid4().hex
-        self.description = kwargs.get('description')
-        self.start = start
-        self.end = end
-        self._switches = []
-        self._links = []
-        self._unis = []
-        self.items = items
-        self.status = kwargs.get('status', Status.PENDING)
-
-    @property
-    def items(self):
-        """Items getter."""
-        return self._switches + self._links + self._unis
-
-    @items.setter
-    def items(self, items):
-        """Items setter."""
-        self._switches = []
-        self._unis = []
-        self._links = []
-        for i in items:
-            if isinstance(i, UNI):
-                self._unis.append(i)
-            elif isinstance(i, Link):
-                self._links.append(i)
-            else:
-                self._switches.append(i)
+    items: conlist(Union[UNI, Link, str], min_items=1)
```

Comment:
@viniarck So I was working to making a model fully compatible with converting from a dict, and I came up with [this](https://gist.github.com/Ktmi/1b7156d7c34a379145879a7e710a415f). This makes it so we can directly convert from our dict/json representation into the Maintanence window object.

However, this will also require updating kytos core to use pydantic models/dataclasses for the id types, as the currently existing ID types aren't fully compatible with pydantic models. The main Issue I encountered was that the string value of the ids were being evaluated before the parameters had been fully parsed, resulting in errors with the LinkID class, and producing the wrong string value for the UNIID class (which I plan to introduce). The string like nature of the current IDs was chosen, because it provides the best backwards compatibility with older code, but it seems like now it may act as a roadblock moving forward.

Reply:
@Ktmi, I think maintenance is trying to model more objects and attributes than it should, as you've highlighted initially, probably just the ids potentially as string and a few more attributes should suffice to store what's needed. It's great to try to reuse existing objects, but when they get in the way and would need significant refactoring when we need to reconsider. I don't think that also introducing new models is an issue in the NApps, it also helps to isolate and clearly represents what needs to be stored without having to worry much about that being changed and leak somewhere else (APIs/interface boundaries) where it might be tried to be reused, so partly duplicating some attributes definitions is OK in this case, but, with that said, I think we can simplify based on your initial idea and only store a subset of things that we need like IDs, scheduler attributes and potentially other minor things that need to be discussed/refined (maybe who created the MW?). Let's have a brainstorm session to chat more about it and then you consolidate the options. 

The intention to use pydantic more in the core is also great, maybe gradually in the future we can try to plan for it.